override/controllers/admin/AdminOrdersController.php

```diff
@@ -134,11 +139,15 @@ public function postProcess()
                     $current_order_state = $order->getCurrentOrderState();
 
                     if ($current_order_state->id != $order_state->id) {
-                        if ($current_order_state !== null && $current_order_state->id != $order_state->id && $current_order_state->name === 'Canceled' && count($order->getOrderPayments())) {
-                            if ($order->payment === 'Comfino payments' && ($current_order_state->paid || in_array($current_order_state->name, $this->comfinoConfirmStates, true))) {
+                        if ($current_order_state !== null && $current_order_state->id != $order_state->id && $order_state->id == Configuration::get('PS_OS_CANCELED') && count($order->getOrderPayments())) {
+                            if ($order->payment === 'Comfino payments' && ($current_order_state->paid || count(array_intersect($current_order_state->name, $this->comfinoConfirmStates)))) {
```

Comment:
```suggestion
                            if ($order->payment === 'Comfino payments' && ($current_order_state->paid || !empty(array_intersect($current_order_state->name, $this->comfinoConfirmStates)))) {
```
aczkolwiek nie spodziewamy się zatrzęsienia wartości do policzenia w tablicy

Reply:
Na początku użyłem !empty, ale zmieniłem kierując się estetyką. Teoretycznie empty jest szybsze.

Reply:
Niby count ma złożoność O(1) dla tablic płaskich:

ZEND_API int zend_hash_num_elements(const HashTable *ht)
{
    IS_CONSISTENT(ht);

    return ht->nNumOfElements;
}